astropy/time/tests/test_quantity_interaction.py

```diff
@@ -160,21 +162,66 @@ def test_valid_quantity_operations2(self):
         g = 10.*u.m/u.second**2
         v = t0 * g
         assert isinstance(v, u.Quantity)
-        assert v.decompose().unit == u.m / u.second
+        assert u.allclose(v, t0.sec * g.value * u.m / u.second)
         q = np.log10(t0/u.second)
         assert isinstance(q, u.Quantity)
         assert q.value == np.log10(t0.sec)
         s = 1.*u.m
         v = s/t0
         assert isinstance(v, u.Quantity)
-        assert v.decompose().unit == u.m / u.second
+        assert u.allclose(v, 1. / t0.sec * u.m / u.s)
+        t = 1.*u.s
+        t2 = t0 * t
+        assert isinstance(t2, u.Quantity)
+        assert u.allclose(t2, t0.sec * u.s ** 2)
         # broadcasting
         t1 = TimeDelta(np.arange(100000., 100012.).reshape(6, 2), format='sec')
         f = np.array([1., 2.]) * u.cycle * u.Hz
         phase = f * t1
         assert isinstance(phase, u.Quantity)
         assert phase.shape == t1.shape
-        assert phase.unit.is_equivalent(u.cycle)
+        assert u.allclose(phase, t1.sec * f.value * u.cycle)
+
+    def test_valid_quantity_operations3(self):
+        """Test a TimeDelta remains one if possible."""
+        t0 = TimeDelta(10., format='jd')
+        q = 10. * u.one
+        t1 = q * t0
+        assert isinstance(t1, TimeDelta)
+        assert t1 == TimeDelta(100., format='jd')
+        t2 = t0 * q
+        assert isinstance(t2, TimeDelta)
+        assert t2 == TimeDelta(100., format='jd')
+        t3 = t0 / q
+        assert isinstance(t3, TimeDelta)
+        assert t3 == TimeDelta(1., format='jd')
+        q2 = 1. * u.percent
+        t4 = t0 * q2
+        assert isinstance(t4, TimeDelta)
+        assert abs(t4 - TimeDelta(0.1, format='jd')) < 1. * u.ns
+        q3 = 1. * u.hr / (36. * u.s)
+        t5 = q3 * t0
+        assert isinstance(t4, TimeDelta)
+        assert abs(t5 - TimeDelta(1000., format='jd')) < 1. * u.ns
+        # Test multiplication with a unit.
+        t6 = t0 * u.one
+        assert isinstance(t6, TimeDelta)
+        assert t6 == TimeDelta(10., format='jd')
+        t7 = u.one * t0
+        assert isinstance(t7, TimeDelta)
+        assert t7 == TimeDelta(10., format='jd')
+        t8 = t0 * ''
```

Comment:
I think this should raise a `TypeError` exception.  Instead it appears the string being coerced to a unit, but that seems like taking duck typing too far in this context where this is no other hint that the string is representing a unit.  If I saw code like `t8 = t0 * 'das'` then I'd be quite perplexed what is going on, but that does work.  (Side note, I was trying to break this and typed a few "random" characters expecting failure, only to discover that `das` is apparently a unit.)

Reply:
This is the one comment I am not sure about (see main thread); do we strictly stick to `Quantity` behaviour, or try to be a bit more logical. My sense is that consistency wins here; I don't think it will actually happen much in practice...

Reply:
OK, consistency wins.